index.d.ts

```diff
@@ -476,6 +492,37 @@ export type ASTNode =
   | BinaryOperation
   | Conditional
   | IndexAccess;
+export type Expression = 
+  | IndexAccess
+  | TupleExpression
+  | BinaryOperation
+  | Conditional
+export type PrimaryExpression = 
+  | BooleanLiteral
+  | NumberLiteral
+  | Identifier
+  | TupleExpression
+  | ElementaryTypeNameExpression
+export type SimpleStatement=
+  | VariableDeclaration
+  | ExpressionStatement
+export type TypeName =
+  | ElementaryTypeName
+  | UserDefinedTypeName
+  | Mapping
+  | FunctionTypeName
+export type Statement =
+  | IfStatement
+  | WhileStatement
+  | ForStatement
+  | Block
+  | InlineAssemblyStatement
+  | DoWhileStatement
+  | ContinueStatement
+  | BreakStatement
+  | ReturnStatement
+  | ThrowStatement
+  | SimpleStatement
```

Comment:
[Need to address] Missing semicolon, better be consistent. Same goes to above union types.